### ISSUE TITLE
Ruin probability aggregates all annual losses into single claim, breaking per-occurrence layer logic

### ISSUE THREAD

**Issue by AlexFiliakov** [AlexFiliakov/Ergodic-Insurance-Limits#1136]
## Problem Statement

`RuinProbabilityAnalyzer._process_simulation_year()` sums all individual loss events for the year into a single `total_loss` and passes it to `InsuranceProgram.process_claim()` as one claim. This fundamentally breaks per-occurrence insurance layer logic: a year with ten $200K losses is treated identically to a year with one $2M loss, even though the insurance response should be very different.

## v1.0 Impact

**High** — Ruin probability is a headline metric of this framework. If the insurance program's claim processing is applied to aggregate losses rather than individual occurrences, the ruin probabilities are systematically biased. Small frequent losses that should stay within the deductible are instead aggregated into amounts that breach higher layers, leading to **overestimated insurance recoveries** for attritional-dominant portfolios and **incorrect ruin probability estimates**.

## Affected Code

- `ergodic_insurance/ruin_probability.py`, lines 448-476 (`_process_simulation_year`)

```python
# Current: aggregates all losses into one claim
total_loss = sum(event.amount for event in events)
claim_result = self.insurance_program.process_claim(total_loss)
```

## Current Behavior

All individual loss events generated for a simulation year are summed into one aggregate amount and processed as a single occurrence through the insurance program. A year with 5 attritional losses of $50K each ($250K total) is processed as a single $250K claim.

## Expected Behavior

Each loss event should be processed individually through `process_claim()`, preserving per-occurrence deductible and layer attachment/limit semantics. The framework already tracks individual `LossEvent` objects, so the fix is to iterate:

```python
total_recovery = 0.0
total_retained = 0.0
for event in events:
    claim_result = self.insurance_program.process_claim(event.amount)
    total_recovery += claim_result.insurance_recovery
    total_retained += event.amount - claim_result.insurance_recovery
```

For aggregate-limit layers, `InsuranceProgram` already tracks annual utilization via `LayerState`, so this change correctly supports both per-occurrence and aggregate structures.

## Actuarial References

- ASOP 43 §3.2 — "The actuary should consider the nature of the coverage (per-occurrence, aggregate, or both) in determining unpaid claim estimates."
- Friedland, *Fundamentals of Actuarial Mathematics* — Distinguishes occurrence-basis from aggregate-basis coverage as fundamental to loss development and reserving methodology.
- CAS Exam 5, Ratemaking Chapter 3 — Per-occurrence vs. aggregate limits produce materially different expected layer losses.
